shards/shards.go

```diff
@@ -225,6 +301,7 @@ func NewDirectorySearcher(dir string) (zoekt.Streamer, error) {
 	}
 
 	go ss.watchPriorities(dir, dw.quit)
+	ss.watchPublic(dir, dw.quit)
```

Comment:
nit: I quite like the decision of watchPriorities being blocking and just calling it in its own goroutine. Makes it easier to reason about and test.

Reply:
I wanted it to be blocking to make sure we have priorities in shardedSearcher for the first search. This is less important for priorities than for public. I could move the blocking part out of watchPublic.